src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.Utf8JsonStreamReader.cs

```diff
@@ -225,6 +227,29 @@ internal static PackageSpec GetPackageSpec(ref Utf8JsonStreamReader jsonReader,
                             }
                         }
                     }
+                    else if (jsonReader.ValueTextEquals(UsingMicrosoftNETSdk))
```

Comment:
This code is reading the root `PackageSpec` json, but that's not where these properties live in the updated PR. Now that it's part of `RestoreMetadata`, you need to move these functions to the appropriate place.

If you write unit tests, this mistake will become very apparent.